### ISSUE TITLE
Exclusive ownership transfer with license back as specified in CLA is concerning

### ISSUE THREAD

**Issue by JonasT** [NodeBB/NodeBB#6115]
The way the CLA works by exclusive ownership transfer with a comprehensive non-exclusive licensing provided back to me appears to be a quite concerninc approach. Why?

Because while for simple changes I might write them exclusively for your project where this is fine, as soon as I want to share them with *multiple* projects I *obviously can't **exclusively** license it to multiple projects at the same time*, that's just not *how things work*. **This means if everyone had a similarly selfish CLA, any contributor would need to limit themselves to just *one* out of all them to contribute any such change ever**, even if otherwise you could easily decide to give it to *all* of them to make the world a better place for everyone. And that's pretty bad outcome for the open-source community as a whole.

Therefore, I suggest you get your lawyers and find a better way that gives you a super comprehensive non-limiting license that allows you to do pretty much anything you want instead of transferring my license to you in a silly lock-out exclusive way which obviously can be only done once with a comprehensive license shared back the other way (which introduces quite silly limitations with how people can help multiple projects that do something similar, or then as a result can't).

Just to make it more obvious: if there ever was a tiny change I might need to contribute a specific substantial change in a similar way to any other projects than just yours ever at any point in the future, *your current license would require me to not contribute it to your project*. It's not that I wouldn't want to, **I literally couldn't just to maintain my own legal safety**. that's because I would never know if someone else would eventually ask for something similarly stupid (an exclusive license transfer) which I then could never comply with since you already took it which would leave me in legal jeopardy. This cannot be seriously in your own interest of collaboration.

TL;DR: I am not a lawyer so obviously I could be very wrong. However, I'm pretty sure I understand the very basic notion of what an *exclusive* license transfer is..

**Comment by julianlam** [comment]
My understanding of the CLA (technically it is a CAA, an _assignment_ agreement) is that you're assigning ownership of the written code (or changeset, etc.) to NodeBB, not the actual concept or idea itself. 

Unless you can find written clauses that say otherwise, there's no reason why our CLA would be forcing you to hand over your right to ever write that line of code again.

... and even if it were, it'd be so unenforceable as to be moot.

We use an assignment agreement so we can more easily license out the software (as we maintain ownership of the code as-is), otherwise we'd have to get permission every time we want to do anything with the code.

**Comment by JonasT** [comment]
> otherwise we'd have to get permission every time we want to do anything with the code.

That's what a comprehensive shared non-retractable license to reuse is for, pretty much what your CLA is giving me back.

> there's no reason why our CLA would be forcing you to hand over your right to ever write that line of code again.

Maybe not a single line of code, but certainly a bigger change set exactly like that since now it's owned by you, especially if someone else wants to *exclusively* own it too.

Again maybe I'm wrong, but it certainly does read pretty bad.

**Comment by JonasT** [comment]
By the way, the Ubuntu CLA reads **entirely different** and as I would expect it to be. For example, it starts out like this:

> You retain ownership of the Copyright in Your
Contribution and have the same rights to use or license the
Contribution which You would have had without entering
into the Agreement.

That's a **completely different** tone than yours which basically starts out like: "Your change is now ours. Completely. Exclusively. Irrevocably. We will now allow you out of the kindness of our heart to reuse it pretty much unrestrictedly but it's still ours now."

**Comment by julianlam** [comment]
Lawyers are a fun bunch, aren't they?

Again, we require assignment of copyright to ease sublicense issues as a withdrawal of licensing by any contributing party would immediately cause any and all code distributed to be in violation of the agreement, at least from our legal counsel's point of view.

That said, the agreement also only covers your contribution, not the idea or implementation, so I don't really read it as us restricting you from ever writing those lines of code ever again in your life, under threat of litigation (that'd be pretty awful for anyone to do, really).

Not to mention we don't even place any limits (assumed or otherwise), as right there in the next paragraph, we hand over those rights (to do anything you want with the code) right back to you.

**Comment by JonasT** [comment]
> only covers your contribution, not the idea or implementation,

That just seems to me like a nonsense thing to say. Obviously it covers what I am contributing, which is **obviously** also the actual code I am writing.

> Not to mention we don't even place any limits (assumed or otherwise)

Actually **you do**, which is my entire point. The *whole notion* of an **exclusive** transfer is that you can *only do it once* because it is **exclusive**. I could of course let others use it with *your* copyright, but I'd be severely limited in how I could do another **exclusive** transfer of it again - which is, pretty much never again. Which is precisely why I'm suggesting what your CLA does is nonsense...

Also sorry if I'm not buying it, but if the Ubuntu CLA can do it better so can you. Ubuntu is a pretty big company and they found a *reasonable* way of doing this. If your legal counsel cannot be convinced of this after reading the Ubuntu CLA, maybe you should change your counsel..

**Comment by JonasT** [comment]
Basically I see what you are saying under an extremely pessimistic, super protective kind of view of your company's side. However, you are throwing your contributors under the bus for a theoretical issue which again Ubuntu (and many others) doesn't seem to see as necessary. Therefore, I suggest you reconsider.

**Comment by julianlam** [comment]
I understand where you are coming from, and how our use of wording in the CLA/CAA is restrictive in your eyes. However, as stated above, we *do* require the exclusivity right purely for ease of sublicensing, and to ensure that we do so in a legally-binding and adherent manner.

With regard to Ubuntu's CLA, one cannot simply compare one with the other as our projects are fundamentally different (if even so far as asking what "distribution" [as a verb] means in the context of a web application) and I cannot in all good conscience adopt it without understanding the intricacies of it. I would also point out that neither our nor Ubuntu's CLAs have been challenged in court, and while I may be wrong on the latter, I (and I would assume, the rest of the NodeBB team) would feel safer with our CLA/CAA wording as-is.

Specifically in relation to the transfer of rights, note section (d) which states that each contributor get a broad non-exclusive license back:

> (d) Upon such transfer of rights to Us, to the maximum extent possible, We immediately grant to You a perpetual, worldwide, non-exclusive, paid-up, royalty-free, transferable, irrevocable license under such rights covering the Contribution, **with rights to sublicense through multiple tiers of sublicensees, to reproduce, modify, display, perform, transmit, make available, distribute and otherwise exploit the Contribution. The intention of the parties is that this license will be as broad as possible and to provide You with rights as similar as possible to the owner of the rights that You transferred**. This license back is limited to the Contribution and does not provide any rights to the Material.

Emphasis mine.

I hope you understand my point of view and why we are hesitant to change our wording at this time. I consider this matter closed (although of course we are all free to discuss and re-open should the need arise).

Your issues have been well-thought-out, researched, and your contributions are much appreciated. I fully understand if signing the CLA/CAA causes you enough concern that it will stop you from contributing, although I do hope that on the other hand, I've assuaged your fears over the use of your contributions, and that you will continue to help improve the NodeBB project over time. 👍

**Comment by JonasT** [comment]
> With regard to Ubuntu's CLA, one cannot simply compare one with the other as our projects are fundamentally different

Well then let's have a few others:

Google CLA: https://cla.developers.google.com/about/google-individual - contributor provides non-exclusive license.

GitHub: https://cla.github.com/agreement - contributor provides non-exclusive license.

Microsoft: https://cla.microsoft.com/cladoc/microsoft-contribution-license-agreement.pdf - contributor provides non-exclusive license.

**Noone** uses the overreaching exclusive complete ownership transfer to the company with non-exclusive back-licensing like you do.

If you are really convinced this is really necessary, all big other companies I checked disagree - which quite frankly makes you look really bad here.

> we do require the exclusivity right purely for ease of sublicensing

Do you though?

> I do hope that on the other hand, I've assuaged your fears over the use of your contributions

I don't mind the use too horribly, otherwise I wouldn't contribute. It does make me vary of substantial contributions but let's be realistic here, I'm not much of a JavaScript developer so I probably wasn't gonna jump in with huge contributions anyway.

However, in the end your CLA remains IMHO poorly thought out and overreaching. You can choose not care but I'm almost certain you are going to lose contributions over this, even if maybe not any from me.

**Comment by pixiuPL** [comment]
Completely agree with @JonasT here.

**Comment by pitaj** [comment]
Are those CLAs for GPLv3 projects?

**Comment by pixiuPL** [comment]
@pitaj.what do you mean by `those '?

**Comment by Pita** [comment]
@pixiuPL I appreciate that ;)

**Comment by Pita** [comment]
Hi @pitaj

**Comment by JonasT** [comment]
@pixiuPL I think this might actually prevent you from submitting your contributions into a different GPL project, since GPL forbids any additional restrictions (and NodeBB arguably forbids submitting your code ever to a similarly licensed project which also wants the exclusive license transfer since due to the nature of exclusivity you can't do that twice, so that *is* effectively a restriction). However, I could be entirely wrong since I'm not a lawyer.

**Comment by pixiuPL** [comment]
@JonasT Thats what I meant......
